src/opendr/perception/face_recognition/face_recognition_learner.py

```diff
@@ -534,6 +538,67 @@ def infer(self, img):
         else:
             raise UserWarning('Infer should be called either with backbone_only mode or with a classifier head')
 
+    def feature_extraction(self, img):
+        """
+        This method is used to perform face recognition on an image.
+
+        :param img: image to run inference on
+        :rtype img: engine.data.Image class object
+        :return: Returns an engine.target.Category object, which holds an ID and the distance between
+                 the embedding of the input image and the closest embedding existing in the reference database.
+        :rtype: engine.target.Category object
+        """
+        closest_id = None
+        if not isinstance(img, Image):
+            img = Image(img)
+        img = img.convert("channels_last", "bgr")
+        if self._model is None and self.ort_backbone_session is None:
+            raise UserWarning('A model should be loaded first')
+        transform = transforms.Compose([
+            transforms.Resize([int(128 * self.input_size[0] / 112), int(128 * self.input_size[0] / 112)]),
+            transforms.CenterCrop([self.input_size[0], self.input_size[1]]),
+            transforms.ToTensor(),
+            transforms.Normalize(mean=self.rgb_mean, std=self.rgb_std)]
+        )
+        if self.mode == 'backbone_only':
+            distance = 100.0
+            if distance == 0:
+                distance = 10.0
+            person = None
```

Comment:
This one is also causing PEP8 to fail:
```
/home/runner/work/opendr/opendr/src/opendr/perception/face_recognition/face_recognition_learner.py:567:13: local variable 'person' is assigned to but never used
```
```suggestion
```

Reply:
We are working with @thomaspeyrucain on this one, so I will ask for a review when it is completed